src/poliastro/bodies.py

```diff
@@ -83,11 +83,22 @@ def __init__(self, parent, k, name, symbol=None, R=0 * u.km):
         self.name = name
         self.symbol = symbol
         self.R = R
+        self.kwargs = kwargs
 
     @classmethod
     @u.quantity_input(k=u.km ** 3 / u.s ** 2, R=u.km)
-    def from_parameters(cls, parent, k, name, symbol, R):
-        return cls(parent, k, name, symbol, R)
+    def from_parameters(cls, parent, k, name, symbol, R, **kwargs):
+        return cls(parent, k, name, symbol, R, **kwargs)
+
+    @classmethod
+    def from_relative(cls, reference, parent=None, k=None, name=None,
+                      symbol=None, R=None, **kwargs):
+        if hasattr(reference, 'k') and hasattr(reference, 'R'):
```

Comment:
I would apply duck typing here instead: I agree with the idea that `reference` does not need to be a `Body`, but do not try to catch the issue. If the `reference` variable does not have `k` or `R`, it is going to raise an `AttributeError` anyway, which I think it's informative enough.